enhancements/sdk-metrics.md

```diff
@@ -0,0 +1,178 @@
+LABEL sdk.operatorframework.io.metadata.layout.v1 = "go.kubebuilder.io/v2.0.0"
+```
+
+Having labels on `bundle.dockerfile` and `annotations.yaml` will help in collecting metrics regarding the use of SDK from indexed operator bundle images.
+
+3. Cluster service version
+
+The cluster service version generated by SDK using the `operator-sdk generate csv` command can have a similar stamp included in the metadata. The stamp will be present as an [annotation][kubernetes_annotation]. For example, while generating a new CSV, the following will be added to Object metadata:
+
+```Go
+func newCSV(name, version string) (*olmapiv1alpha1.ClusterServiceVersion, error) {
+    csv := &olmapiv1alpha1.ClusterServiceVersion{
+      ObjectMeta: metav1.ObjectMeta{
+        Name:      getCSVName(name, version),
+        Namespace: "placeholder",
+        Annotations: map[string]string{
+          "capabilities": "Basic Install",
+          "sdk.operatorframework.io/version/v1": "v0.17.0",
+          "sdk.operatorframework.io/operator-type/v1": "go.kubebuilder.io/v2.0.0",
+		}
+    }
+    ...
+  }
+}
+```
+
+The corresponding CSV will appear to be:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    capabilities: Basic Install
+    sdk.operatorframework.io/version/v1: "v0.17.0"
+    sdk.operatorframework.io/operator-type/v1: "go.kubebuilder.io/v2.0.0"
+...
+```
+
+In case of legacy projects, which do not use Kubebuilder layout, the `sdk.operatorframework.io/operator-type` annotation will only represent the kind of operator (go/ansible/helm). For example, the annotations would be:
+
+```Go
+Annotations: map[string]string{
+  "capabilities": "Basic Install",
+  "sdk.operatorframework.io/version/v1": "v0.15.0",
+  "sdk.operatorframework.io/operator-type/v1": "go",
+},
+```
+
+4. Custom Resource Definition
+
+As OLM would be moving towards having CSV-less bundles, generation of CSV would not be performed by SDK. Injecting stamps on CRDs would be helpful to propagate the metadata regarding SDK through the bundles.
+
+The CRD manifest generated using kustomize templates will have similar annotations in the metadata. For example, the metadata in CRD would look like:  
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: memcacheds.cache.example.com
+    annotations:
+      sdk.operatorframework.io/version/v1: "v0.17.0"
+      sdk.operatorframework.io/operator-type/v1: "go.kubebuilder.io/v2.0.0"
+```
+
+#### Integration of SDK stamps with OLM metrics
+
+The main aim of adding SDK stamps to bundle images and CSV is to embed SDK related data with OLM metrics. Currently, OLM tracks the lifecycle of CSV and reports metrics related to its success or failure. The first step, would be to include SDK metadata in those metrics by adding labels. The two relevant metrics where we can the label currently are - `csvSucceeded` and `csvAbnormal`.
```

Comment:
I suppose we can skip `csvCount` (which is Number of CSVs successfully registered).